public/app/plugins/panel/graph/graph.ts

```diff
@@ -100,10 +100,11 @@ class GraphElement {
     const graphHeight = this.elem.height();
     updateLegendValues(this.data, this.panel, graphHeight);
 
+    if (this.legendElem.hasChildNodes()) {
+      ReactDOM.unmountComponentAtNode(this.legendElem);
```

Comment:
why unmount on every render? must be another way to fix this

Reply:
> why unmount on every render? must be another way to fix this

ok, i will try another solution.

Reply:
Do you know when you can do that?

Reply:
Yes, I've changed the solution, but I'm not sure that's a good solution.